lib/styles.js

```diff
@@ -347,4 +347,118 @@ module.exports = function (expect) {
+        for (var lineNumber = 0 ; lineNumber < maxHeight ; lineNumber += 1) {
+            if (lineNumber > 0) {
+                this.nl();
+            }
+            var i;
+            for (i = 0 ; i < blockNumbers.length ; i += 1) {
+                blockNumbers[i] = 0;
+                blockOffsets[i] = 0;
+            }
+            var contentLeft;
+            do {
+                contentLeft = false;
+                var hasOutputChar = false;
+                for (i = 0 ; i < flattenedPens.length ; i += 1) {
+                    var currentLine = flattenedPens[i][lineNumber];
+                    if (currentLine) {
+                        while (currentLine[blockNumbers[i]] && blockOffsets[i] >= currentLine[blockNumbers[i]].args.content.length) {
+                            blockNumbers[i] += 1;
+                            blockOffsets[i] = 0;
+                        }
+                        var currentBlock = currentLine[blockNumbers[i]];
+                        if (currentBlock) {
+                            contentLeft = true;
+                            if (!hasOutputChar) {
+                                var ch = currentBlock.args.content.charAt(blockOffsets[i]);
+                                if (ch !== ' ') {
+                                    this.text(ch, currentBlock.args.styles);
+                                    hasOutputChar = true;
+                                }
+                            }
+                            blockOffsets[i] += 1;
+                        }
+                    }
+                }
+                if (!hasOutputChar && contentLeft) {
+                    this.sp();
+                }
+            } while (contentLeft);
+        }
+    });
+
+    expect.addStyle('arrowsAlongsideChangeOutputs', function (packing, changeOutputs) {
+        if (packing) {
+            var topByChangeNumber = {};
+            var top = 0;
+            changeOutputs.forEach(function (changeOutput, index) {
+                topByChangeNumber[index] = top;
+                top += changeOutput.size().height;
+            });
+            var that = this;
+
+            var arrows = [];
+            packing.forEach(function (columnSet, i, packing) {
+                columnSet.forEach(function (entry) {
+                    arrows.push(that.clone().arrow({
+                        left: i * 2,
+                        top: topByChangeNumber[entry.start],
+                        width: 1 + (packing.length - i) * 2,
+                        height: topByChangeNumber[entry.end] - topByChangeNumber[entry.start] + 1,
+                        direction: entry.direction
+                    }));
+                });
+            });
+
+            if (arrows.length === 1) {
+                this.block(arrows[0]);
+            } else if (arrows.length > 1) {
+                this.block(function () {
+                    this.merge(arrows);
+                });
+            }
```

Comment:
The way `merge` is implemented, it takes an array of pens and produces a fresh pen with the merged output. It does handle the single pen case, but I'd like to avoid creating a copy here. My initial attempt at implementing the merging was to merge one or more other pens into the current pen, but that turned out to be harder to implement and not as understandable.